indico/cli/i18n.py

```diff
@@ -84,25 +91,153 @@ def cli():
 }
 
 
-def find_packages(path, prefix=''):
+def _get_plugin_options(cmd_name, plugin_dir):
+    return {
+        'init_catalog': {
+            'input_file': _get_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'extract_messages': {
+            'output_file': _get_messages_pot(plugin_dir),
+            'mapping_file': 'babel.cfg' if os.path.isfile(os.path.join(plugin_dir, 'babel.cfg')) else '../babel.cfg'
+        },
+        'compile_catalog': {
+            'directory': _get_translations_dir(plugin_dir)
+        },
+        'update_catalog': {
+            'input_file': _get_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+
+        # JavaScript
+        'init_catalog_js': {
+            'input_file': _get_js_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'extract_messages_js': {
+            'keywords': '$t gettext ngettext:1,2 pgettext:1c,2 npgettext:1c,2,3',
+            'output_file': _get_js_messages_pot(plugin_dir),
+            'mapping_file': 'babel-js.cfg' if os.path.isfile(os.path.join(plugin_dir, 'babel-js.cfg'))
+                            else '../babel-js.cfg'
+        },
+        'update_catalog_js': {
+            'input_file': _get_js_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+
+        # JavaScript / React
+        'init_catalog_react': {
+            'input_file': _get_react_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'update_catalog_react': {
+            'input_file': _get_react_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+    }[cmd_name]
+
+
+@contextmanager
+def _chdir(path):
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
+def _validate_plugin_dir(plugin_dir):
+    if not os.path.exists(os.path.join(plugin_dir, 'setup.py')):
+        raise click.BadParameter(f'no setup.py found in {plugin_dir}')
+    for dirs in os.listdir(plugin_dir):
+        if (os.path.isdir(os.path.join(plugin_dir, dirs))
+                and 'translations' in os.listdir(os.path.join(plugin_dir, dirs))):
+            break
+    else:
+        raise click.BadParameter(f'no translations folder found in {plugin_dir}')
+    return plugin_dir
+
+
+def _validate_plugin_dirs(ctx, param, values):
+    for value in values:
+        _validate_plugin_dir(value)
+    return values
+
+
+def _validate_all_plugins_dir(ctx, param, value):
+    plugin_dirs = []
+    for dirs in os.listdir(value):
+        try:
+            plugin_dirs.append(_validate_plugin_dir(os.path.join(value, dirs)))
+        except click.BadParameter:
+            pass
```

Comment:
Considering that this function never modifies the value, I don't particularly like that it returns it and that return value is used... How about something like this instead?

```
for ...:
    plugin_dir = ...
    try: _validate(plugin_dir)
    except ...: continue
    plugin_dirs.append(plugin_dir)
```

Reply:
I've changed this to no longer collect the return value. Also the _validate_plugin_dir no longer returns anything.